File: python/1038.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:

    def bstToGst(self, root: TreeNode) -> TreeNode:
        s = 0
        def helper(root: TreeNode) -> None:
            if root:
                nonlocal s
                helper(root.right)
                root.val += s
                s = root.val
                helper(root.left)
        helper(root)
        return root

File: go/1038.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func bstToGst(root *TreeNode) *TreeNode {
	var s int
	var helper func(*TreeNode)
	helper = func(root *TreeNode) {
		if root != nil {
			helper(root.Right)
			s += root.Val
			root.Val = s
			helper(root.Left)
		}
	}
	helper(root)
	return root
}
